File: Sample Code/MonoGame/Rendering - Animated Sprites/Rendering - Animated Sprites/Rendering - Animated Sprites/Objects/Bird.cs
using Microsoft.Xna.Framework;

namespace CS5410.Objects
{
    public class Bird : AnimatedSprite
    {
        private readonly double m_moveRate;
        private readonly double m_rotateRate;

        public Bird(Vector2 size, Vector2 center, double moveRate, double rotateRate) : base(size, center)
        {
            m_moveRate = moveRate;
            m_rotateRate = rotateRate;
        }

        public void moveForward(GameTime gameTime)
        {
            //
            // Create a normalized direction vector
            double vectorX = System.Math.Cos(m_rotation);
            double vectorY = System.Math.Sin(m_rotation);
            //
            // With the normalized direction vector, move the center of the sprite
            m_center.X += (float)(vectorX * m_moveRate * gameTime.ElapsedGameTime.TotalMilliseconds);
            m_center.Y += (float)(vectorY * m_moveRate * gameTime.ElapsedGameTime.TotalMilliseconds);
        }

        public void rotateLeft(GameTime gameTime)
        {
            m_rotation -= (float)(m_rotateRate * gameTime.ElapsedGameTime.TotalMilliseconds);
        }

        public void rotateRight(GameTime gameTime)
        {
            m_rotation += (float)(m_rotateRate * gameTime.ElapsedGameTime.TotalMilliseconds);
        }
    }
}

File: Sample Code/JavaScript/SpriteSheet - Bird/SpriteSheet - Bird/scripts/objects/bird.js
//------------------------------------------------------------------
//
// Creates a Bird model based upon the passed in specification.
//
//------------------------------------------------------------------
MyGame.objects.Bird = function(spec) {

    //------------------------------------------------------------------
    //
    // Move in the direction of the rotation.
    //
    //------------------------------------------------------------------
    function moveForward(elapsedTime) {
            //
            // Create a normalized direction vector
            let vectorX = Math.cos(spec.rotation);
            let vectorY = Math.sin(spec.rotation);
            //
            // With the normalized direction vector, move the center of the sprite
            spec.center.x += (vectorX * spec.moveRate * elapsedTime);
            spec.center.y += (vectorY * spec.moveRate * elapsedTime);
    }

    function rotateLeft(elapsedTime) {
        spec.rotation -= spec.rotateRate * (elapsedTime);
    }

    function rotateRight(elapsedTime) {
        spec.rotation += spec.rotateRate * (elapsedTime);
    }

    let api = {
        get size() { return spec.size; },
        get center() { return spec.center; },
        get rotation() { return spec.rotation; },
        moveForward: moveForward,
        rotateLeft: rotateLeft,
        rotateRight: rotateRight
    };

    return api;
};
